OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnPrerequisite.cs

```diff
@@ -62,6 +61,11 @@ void INotifyRemovedFromWorld.RemovedFromWorld(Actor self)
 				globalManager.Unregister(self, this, info.Prerequisites);
 		}
 
+		void INotifyOwnerChanged.OnOwnerChanged(Actor self, Player oldOwner, Player newOwner)
+		{
+			globalManager = newOwner.PlayerActor.Trait<GrantConditionOnPrerequisiteManager>();
```

Comment:
Unlike `GrantConditionOnPrerequisiteManager.Register`, `GrantConditionOnPrerequisiteManager.Unregister` does not call `PrerequisitesUpdated`. I already accounted for `INotifyRemovedFromWorld.RemovedFromWorld` and `INotifyAddedToWorld.AddedToWorld`. It is `GrantConditionOnPrerequisiteManager.Register` that fixes it after all.